js/controllers/newsliController.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+(function (exports) {
+  exports.NewsliController = function (ArticleList, ArticleListView, ArticleView) {
+    articleList = new ArticleList;
+    articleListView = new ArticleListView(articleList)
+    articleView = new ArticleView;
+
+    articlesContainer = document.getElementById("articles-view");
+    articleContainer = document.getElementById("single-article-view");
+    imageNodes = document.getElementsByTagName("IMG")
+    backButton = "<button id='back' type='button'>Back<button>"
+
+    makersDomain = 'http://news-summary-api.herokuapp.com/';
+    guardianAPIRequest = 'guardian?apiRequestUrl='
+    url = 'http://content.guardianapis.com/search?'
+    allFields = 'show-fields=all';
+
+    function loadContent() {
+      xhttp = new XMLHttpRequest();
+      xhttp.onreadystatechange = function () {
+        if (this.readyState == 4 && this.status == 200) {
+          articles = JSON.parse(this.responseText).response.results;
+          updateArticleList(articles)
+          showSingleArticle()
+        };
+      };
+      xhttp.open("GET", makersDomain + guardianAPIRequest + url + allFields, true);
+      xhttp.send()
+    }
+
+    function showSingleArticle() {
+      imageArray = [].slice.call(imageNodes)
```

Comment:
Took me a while to follow the flow here. The code itself is clean and readable which is great, but I think that you're trying to do too many things in the same class. You mentioned in your self-review about routes and I think starting with routes would actually help you here, as it would help you separate concerns right from the start - you can then think of your code as sort of a pipeline where the various segments are:

- Parse the route
- Match the route to a controller method
- Build the model
- Inject the model into the view
- Render the view into the page

Each step is a different concern and can be handled by a different class. If you try and think about how to do each one on it's own it should then be clearer how to build an effective pipeline.